src/Instrumentation/Yii/src/YiiInstrumentation.php

```diff
@@ -0,0 +1,160 @@
+                }
+                $scope->detach();
+
+                $span = Span::fromContext($scope->context());
+
+                if ($response) {
+                    $statusCode = $response->getStatusCode();
+                    $span->setAttribute(TraceAttributes::HTTP_RESPONSE_STATUS_CODE, $statusCode);
+                    $span->setAttribute(TraceAttributes::NETWORK_PROTOCOL_VERSION, $response->version);
+                    $span->setAttribute(TraceAttributes::HTTP_RESPONSE_BODY_SIZE, YiiInstrumentation::getResponseLength($response));
+
+                    $headers = $response->getHeaders();
+
+                    foreach ((array) (get_cfg_var('otel.instrumentation.http.response_headers') ?: []) as $header) {
+                        if ($headers->has($header)) {
+                            /** @psalm-suppress ArgumentTypeCoercion */
+                            $span->setAttribute(sprintf('http.response.header.%s', strtr(strtolower($header), ['-' => '_'])), $headers->get($header, null, true));
+                        }
+                    }
+
+                    if ($statusCode >= 400 && $statusCode < 600) {
+                        $span->setStatus(StatusCode::STATUS_ERROR);
+                    }
+
+                    // Propagate traceresponse header to response, if TraceResponsePropagator is present
+                    if (class_exists('OpenTelemetry\Contrib\Propagation\TraceResponse\TraceResponsePropagator')) {
+                        /** @phan-suppress-next-line PhanUndeclaredClassMethod */
+                        $prop = new \OpenTelemetry\Contrib\Propagation\TraceResponse\TraceResponsePropagator();
+                        /** @phan-suppress-next-line PhanUndeclaredClassMethod */
+                        $prop->inject($response, ResponsePropagationSetter::instance(), $scope->context());
+                    }
+                }
+
+                if ($exception) {
+                    $span->recordException($exception, [TraceAttributes::EXCEPTION_ESCAPED => true]);
+                    $span->setStatus(StatusCode::STATUS_ERROR, $exception->getMessage());
+                }
+
+                $span->end();
+            }
+        );
+
+        hook(
+            Controller::class,
+            'beforeAction',
+            pre: static function (
+                Controller $controller,
+                array $params,
+                string $class,
+                string $function,
+                ?string $filename,
+                ?int $lineno,
+            ) : void {
+                $action = $params[0] ?? null;
+                $scope = Context::storage()->scope();
+                if (!$action || !$scope) {
+                    return;
+                }
+
+                $span = Span::fromContext($scope->context());
+                $route = YiiInstrumentation::normalizeRouteName(get_class($controller), $action->actionMethod);
+                $span->setAttribute(TraceAttributes::HTTP_ROUTE, $route);
```

Comment:
https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name says we should update the span name here

Reply:
Done and updated tests.